legal-api/src/legal_api/services/involuntary_dissolution.py

```diff
@@ -0,0 +1,168 @@
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This provides the service for involuntary dissolution."""
+import pytz
+from datedelta import datedelta
+from sqlalchemy import exists, not_
+
+from legal_api.models import Batch, BatchProcessing, Business, Filing, db
+from legal_api.utils.datetime import datetime
+
+
+class InvoluntaryDissolutionService():
+    """Provides services to get information for involuntary dissolution."""
+
+    @staticmethod
+    def get_businesses_eligible_count():
+        """Return the number of businesses eligible for involuntary dissolution."""
+        eligible_types = [
+            Business.LegalTypes.COMP.value,
+            Business.LegalTypes.BC_ULC_COMPANY.value,
+            Business.LegalTypes.BC_CCC.value,
+            Business.LegalTypes.BCOMP.value,
+            Business.LegalTypes.CONTINUE_IN.value,
+            Business.LegalTypes.ULC_CONTINUE_IN.value,
+            Business.LegalTypes.CCC_CONTINUE_IN.value,
+            Business.LegalTypes.BCOMP_CONTINUE_IN.value,
+            Business.LegalTypes.EXTRA_PRO_A.value,
+            Business.LegalTypes.LIMITED_CO.value
+        ]
+
+        subquery = exists().where(BatchProcessing.business_id == Business.id,
+                                  BatchProcessing.status != 'WITHDRAWN',
+                                  BatchProcessing.batch_id == Batch.id,
+                                  Batch.batch_type == 'INVOLUNTARY_DISSOLUTION')
+
+        query = db.session.query(Business).\
+            filter(Business.state == Business.State.ACTIVE).\
+            filter(Business.legal_type.in_(eligible_types)).\
+            filter(Business.no_dissolution.is_(False)).\
+            filter(not_(subquery))
+
+        results = query.all()
+
+        seleted_count = 0
+        for business in results:
+            selected = False
+            # selection criteria
+            if _has_specific_filing_overdue(business):
+                selected = True
+            if _has_no_transition_filed_after_restoration(business):
+                selected = True
+            # exclusion criteria
+            if _has_future_effective_filing(business):
+                continue
+            if _has_change_of_address_filing(business):
+                continue
+            if _has_delay_of_dissolution_filing(business):
+                continue
+            if _is_xpro_from_nwpta(business):
+                continue
+            if selected:
+                seleted_count += 1
```

Comment:
I think we want to have the db do as much of the work as possible.  As of mid feb 2023, there were over 30K businesses that were eligible for dissolution so that would be super heavy.

Can we try checking all these things as part of the sqlalchemy query?